modules/tinymce/src/core/main/ts/keyboard/PreventNoneditableInput.ts

```diff
@@ -0,0 +1,12 @@
+import Editor from '../api/Editor';
+
+export const setup = (editor: Editor): void => {
+  editor.on('BeforeInput', (e) => {
+    // Normally input is blocked on non-editable elements that have contenteditable="false" however we are also treating
+    // SVG elements as non-editable and Firefox lets you delete parts of the SVG if you manage to select a part. So this
+    // checks for that scenario and prevents the default behaviour.
+    if (!editor.selection.isEditable()) {
+      e.preventDefault();
```

Comment:
could we potentially remove other checks from the code because this now blocks those scenarios?

Reply:
We might but might be harder to test those things in isolation.

So theoretically we could get rid of some code but that might then make it harder to develop and test those things in isolation if you want the code to be non-editable aware. This just checks for the scenario of when the selection is within a non-editable context but there are other scenarios around non-editables where the caret is in a editable context but say deleting into a non-editable structure this wouldn't be blocked by this.

Reply:
I think reducing how much individual code needs to be aware of non-editable is good, so the ways we check for it are centralised. Just look at how many places you've had to update this year to stabilise non-editable sections.

Reply:
Yes but bit this change doesn't do that. What would do that is to have a centralized dom model api that Is non editable aware. But at least now we have centralized way of checking if something is editable that is editor centric not dom centric only so that it could base it result on options and schema rules. So slowly we are moving towards better apis here.

Reply:
ok. I do want a centralised dom model api, for the record 😂